src/vm/wks/CMakeLists.txt

```diff
@@ -87,21 +29,21 @@ if (WIN32)
         message(FATAL_ERROR "POWERSHELL not found")
     endif()
 
+    # Get the current list of definitions
+    get_compile_definitions(DEFINITIONS)
     add_custom_command(
         OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/AsmConstants.inc"
         DEPENDS ${VM_DIR}/${ARCH_SOURCES_DIR}/asmconstants.h
         COMMAND ${POWERSHELL} -NoProfile -ExecutionPolicy Bypass -NonInteractive \"& \"\"${VM_DIR}/h2inc.ps1\"\"\" \"\"\"${VM_DIR}/${ARCH_SOURCES_DIR}/asmconstants.h\"\"\" >"${CMAKE_CURRENT_BINARY_DIR}/AsmConstants.tmp"
         COMMAND ${CMAKE_CXX_COMPILER} ${DEFINITIONS} /EP "${CMAKE_CURRENT_BINARY_DIR}/AsmConstants.tmp" >"${CMAKE_CURRENT_BINARY_DIR}/AsmConstants.inc"
     )
 
-    set_source_files_properties(${CMAKE_CURRENT_BINARY_DIR}/AsmConstants.inc PROPERTIES GENERATED TRUE)
-
-    add_custom_target(
```

Comment:
Is the custom target really not needed anymore to ensure proper dependency handling?

Reply:
By adding the file as an input to the cee_wks target, we don't need the custom target to ensure proper dependency handling.

Reply:
This didn't work for header files. But maybe the new cmake fixed that.

Reply:
The HEADER_FILE_ONLY source property is what enables this. We can add it to the target as a file so it is dependency-checked, but it is not added to the compilation command itself.

Reply:
I know about this property, but it was still not working with it. I believe it was a known problem hit by various people. Maybe they've really fixed that in the recent cmake. You usually don't spot that it doesn't work, because you seldom touch the AsmConstants.h. Can you try to make a build, then touch the AsmConstants.h and see if the assembler sources that include asmconstants.inc get rebuilt?

Reply:
Will do.

Reply:
You are correct. I'll swap this back to being a custom target.